### ISSUE TITLE
awesome repo! - couple of questions

### ISSUE THREAD

**Issue by rismoney** [davidlublink/vcenter-csi-driver#1]
This project is awesome!  I was exploring it and had a couple of questions.
With regard to the volumes, this will automatically create VMDK files on the datastore? Are those added to the VM running nomad, or are they connected via ISCSI inside the VM or how does that actually work?
These volumes need to be formatted, or is that automatically handled.

I have only worked with a NFS CSI implementation, that effectively just required a share, and it "worked"
I have all the pre-reqs ready to go to test this, so I am very interested in helping this project out.

**Comment by davidlublink** [comment]
I added information to the project readme about life cycles that contains answers to all your questions.

tl;dr;

> this will automatically create VMDK files on the datastore

Yes! However the naming is a bit broken on my setup and I don't know if it's me or not yet. I opened an issue with vmware. Your disks may appear in fcd/uuid.vdmk instead of the folder you specify.

> Are those added to the VM running nomad

Only as needed. During volume creation, it is temporarily connected for creation of ext4 file system. It's disconnected once the filesystem is created.

> are they connected via ISCSI inside the VM
Exactly! The plugin is connecting the disk as an iSCSI device so it appears in /dev/sdX. This way it's high performance and the plugin has nothing else to do.

>  how does that actually work?

This entire plugin is literally 3 API calls being 'translated' from CSI to pyvmomi. Nomad and Vmware are doing 99% of the work.

> These volumes need to be formatted, or is that automatically handled.
ext4 right now. Maybe I can add a parameter to control this, not sure if anyone cares

> I have only worked with a NFS CSI implementation, that effectively just required a share, and it "worked"
I haven't worked with any CSI implementations other than this one, but I was surprised at how easily it 'just worked'. 

> I have all the pre-reqs ready to go to test this, so I am very interested in helping this project out.

Let me know if you have any other questions that might prevent you from testing.

**Comment by rismoney** [comment]
Thank you so much for the quick reply!  I will close this, and if I have any further questions or changes, I'll open a PR or create an issue.

**Comment by davidlublink** [comment]
Don't hesitate to poke me if you have questions. I am also interested in any tests you do, even if they all succeed without issue.
